rpcs3/Emu/RSX/Common/texture_cache.h

```diff
@@ -1072,30 +1059,42 @@ namespace rsx
 			return nullptr;
 		}
 
-		template <typename ...Args>
-		void lock_memory_region(image_storage_type* image, const address_range &rsx_range, u32 width, u32 height, u32 pitch, Args&&... extras)
+		template <typename ...FlushArgs, typename ...Args>
+		void lock_memory_region(image_storage_type* image, const address_range &rsx_range, u32 width, u32 height, u32 pitch, const std::tuple<FlushArgs...>& flush_extras, Args&&... extras)
 		{
 			AUDIT( g_cfg.video.write_color_buffers ); // this method is only called when WCB is enabled
 
 			std::lock_guard lock(m_cache_mutex);
 
+			// Find a cached section to use
 			section_storage_type& region = *find_cached_texture(rsx_range, true, false);
 
-			if (region.get_context() != texture_upload_context::framebuffer_storage && region.exists())
+			if (!region.is_locked())
 			{
+				// Invalidate sections from surface cache occupying same address range
+				std::apply(&texture_cache::invalidate_range_impl_base<FlushArgs...>, std::tuple_cat(std::make_tuple(this, rsx_range, invalidation_cause::superseded_by_fbo), flush_extras));
```

Comment:
I'll let c++ people take a look at this, just in case there is a cleaner way of passing the varags